src/shared_strings.c

```diff
@@ -43,13 +49,13 @@ _new_sst()
     /* Initialise the order list. */
     STAILQ_INIT(sst->order_list);
 
-    /* Store the number of buckets to calculate the load factor. */
-    sst->num_buckets = NUM_SST_BUCKETS;
+    /* Initialize the RB tree. */
+    RB_INIT(sst->rb_tree);
 
     return sst;
 
 mem_error2:
```

Comment:
Not your issue but I wonder if we still need 2 mem_error labels since `free(sst)` should take care of it. You could fix that in passing or I can look into it.